### ISSUE TITLE
Addon 1.0.1 keeps crashing - HA 2022.2.7

### ISSUE THREAD

**Issue by raidolo** [allenporter/stream-addons#33]
Hi, 

I got a sricam rtsp camera, my config is pretty simple:

```
camera:
  - platform: generic
    name: Sricam
    stream_source: rtsp://192.168.1.50/onvif1
```

The addon keeps crashing when I open the stream from the RTSPtoWebRTC integration in HA. 

Any advice?

Here is the log from the addon: 

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[14:04:02] INFO: Starting discovery service
[14:04:02] INFO: Starting RSTPtoWebRTC
time="2022-02-16T14:04:02+01:00" level=info msg="Server CORE start" func=main module=main
time="2022-02-16T14:04:02+01:00" level=info msg="Server RTSP start" call=Start func=RTSPServer module=rtsp_server
time="2022-02-16T14:04:02+01:00" level=info msg="Server HTTP start" call=Start func=RTSPServer module=http_server
time="2022-02-16T14:04:02+01:00" level=info msg="Server start success a wait signals" func=main module=main
[14:04:03] INFO: Successfully send discovery information to Home Assistant.
time="2022-02-16T14:04:18+01:00" level=error msg="stream not found" call=StreamChannelExist channel=0 func=HTTPAPIServerStreamHLSM3U8 module=http_hls stream=camera.sricam
time="2022-02-16T14:04:19+01:00" level=error msg="stream not found" call=StreamChannelExist channel=0 func=HTTPAPIServerStreamHLSM3U8 module=http_hls stream=camera.sricam
time="2022-02-16T14:04:49+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:04:51+01:00" level=info msg="Success connection RTSP" call=Start channel=0 func=StreamServerRunStream module=core stream=camera.sricam
time="2022-02-16T14:04:51+01:00" level=error msg="Stream error restart stream stream stop rtsp signal" call=Restart
time="2022-02-16T14:04:51+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:04:53+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:04:53+01:00" level=info msg="Success connection RTSP" call=Start channel=0 func=StreamServerRunStream module=core stream=camera.sricam
time="2022-02-16T14:04:53+01:00" level=error msg="Stream error restart stream stream stop rtsp signal" call=Restart
time="2022-02-16T14:04:54+01:00" level=error msg="Stream error restart stream read tcp 192.168.1.30:53574->192.168.1.50:554: read: connection reset by peer" call=Restart
time="2022-02-16T14:04:55+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:04:56+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:04:57+01:00" level=info msg="Success connection RTSP" call=Start channel=0 func=StreamServerRunStream module=core stream=camera.sricam
time="2022-02-16T14:04:57+01:00" level=error msg="Stream error restart stream stream stop rtsp signal" call=Restart
time="2022-02-16T14:04:58+01:00" level=info msg="Success connection RTSP" call=Start channel=0 func=StreamServerRunStream module=core stream=camera.sricam
time="2022-02-16T14:04:59+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:04:59+01:00" level=info msg="Success connection RTSP" call=Start channel=0 func=StreamServerRunStream module=core stream=camera.sricam
time="2022-02-16T14:05:01+01:00" level=error msg="stream no video" call=ErrorStreamNoVideo channel=0 func=HTTPAPIServerStreamWebRTC module=http_webrtc stream=camera.sricam
time="2022-02-16T14:05:28+01:00" level=error msg="WebRTC Client Offline" call=WritePacket channel=0 func=HTTPAPIServerStreamWebRTC module=http_webrtc stream=camera.sricam
time="2022-02-16T14:05:37+01:00" level=info msg="Run stream" call=Run channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
time="2022-02-16T14:05:37+01:00" level=info msg="Stream exit by signal or not client" call=StreamServerRunStream channel=0 func=StreamServerRunStreamDo module=core stream=camera.sricam
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xba062b]
goroutine 53 [running]:
main.(*MuxerHLS).SetFPS(...)
	/build/github.com/deepch/RTSPtoWeb/hlsMuxer.go:43
main.(*StorageST).HlsMuxerSetFPS(0xc000434000, 0xc00035657d, 0xd, 0x12a7380, 0x1, 0xf)
	/build/github.com/deepch/RTSPtoWeb/storageStreamChannel.go:353 +0x20b
main.StreamServerRunStream(0xc00035657d, 0xd, 0x12a7380, 0x1, 0xc0000c28f0, 0x0, 0x0, 0x0)
	/build/github.com/deepch/RTSPtoWeb/streamCore.go:173 +0xa98
main.StreamServerRunStreamDo(0xc00035657d, 0xd, 0x12a7380, 0x1)
	/build/github.com/deepch/RTSPtoWeb/streamCore.go:43 +0x4fa
created by main.(*StorageST).StreamAdd
	/build/github.com/deepch/RTSPtoWeb/storageStream.go:35 +0x3bf
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by allenporter** [comment]
Hi, can you file this bug in the upstream project RTSPtoWeb? This project is for the add-on in home assistant but not the actual server.

**Comment by raidolo** [comment]
Sure!

**Comment by raidolo** [comment]
Here you go: https://github.com/deepch/RTSPtoWeb/issues/89

**Comment by allenporter** [comment]
👍🏼 thank you.

**Comment by allenporter** [comment]
I see the author submitted a fix, that was fast. We need to build a new addon image to pick it up so leaving this open to track that.

**Comment by raidolo** [comment]
That was fast... thanks! I can test it then...

**Comment by allenporter** [comment]
The upstream fix for this will be included in 1.1.0

**Comment by raidolo** [comment]
I still got the segfault, i've commented again in the RTSPtoWeb issue that was closed....
